notion-core/src/test/scala/com/ambiata/notion/core/LocationIOSpec.scala

```diff
@@ -120,6 +122,18 @@ class LocationIOSpec extends Specification with ScalaCheck { def is = s2"""
     e <- i.exists(p)
   } yield e ==== true)
 
+  def copy = prop((from: LocationTemporary, to: LocationTemporary, bytes: Array[Byte]) =>
+    for {
+      f   <- from.location
+      i   <- from.io
+      s1   = new String(bytes, "UTF-8")
```

Comment:
This is my fault, I confused the situation with a previous comment, but this isn't what I meant. This is going to break (well at least the other one that test the symmetric nature, random bytes encoded to strings aren't symmetric). What I was getting at with previous comment is that strings are irrelevant for copying a file, we really need to get out of the particularly dangerous habit of believing all content is a string - we need to be checking that copy is an exact byte-for-byte copy, there should be no strings in any of these tests, at best it is irrelevant and at worst it hides very dangerous corner cases that will lead to corruption.